projects/storefrontlib/src/shared/services/trigger/config/trigger.service.ts

```diff
@@ -1,28 +1,61 @@
-import { Injectable, ComponentFactory } from '@angular/core';
+import {
+  Injectable,
+  ComponentFactory,
+  ComponentFactoryResolver,
+} from '@angular/core';
 import {
   TriggerOutletMapping,
   TriggerInlineMapping,
   TriggerUrlMapping,
+  TriggerConfig,
+  TRIGGER_CALLER,
+  TriggerMapping,
 } from './trigger-config';
 import { OutletService } from '../../../../cms-structure/outlet/index';
 
 @Injectable({ providedIn: 'root' })
 export class TriggerService {
-  constructor(protected outletService: OutletService<ComponentFactory<any>>) {}
+  // Keep a list of rendered elements
+  protected renderedCallers: TRIGGER_CALLER[] = [];
 
-  renderDialog(
-    config: TriggerOutletMapping | TriggerInlineMapping | TriggerUrlMapping,
-    template?: ComponentFactory<any>
-  ): void | string {
+  constructor(
```

Comment:
This service gets fairly busy. It's not a big deal for now, but we could consider bringing in different strategies to render the target. If we have those decoupled, the outletService would only be a dependency for the `OutletRenderStrategy`. 
Just food for thought.